be/src/vec/aggregate_functions/aggregate_function_approx_topn.h

```diff
@@ -0,0 +1,286 @@
+            }
+
+            sub_writer.Uint64(result.count);
+            for (size_t i = 0; i < argument_columns.size(); i++) {
+                sub_writer.EndObject();
+            }
+            writer.RawValue(sub_buffer.GetString(), sub_buffer.GetSize(), rapidjson::kObjectType);
+        }
+        writer.EndArray();
+        std::string res = buffer.GetString();
+        data_to.insert_data(res.data(), res.size());
+    }
+
+private:
+    void lazy_init(const IColumn** columns, ssize_t row_num) const {
+        auto get_param = [](size_t idx, const DataTypes& data_types,
+                            const IColumn** columns) -> uint64_t {
+            const auto& data_type = data_types.at(idx);
+            const IColumn* column = columns[idx];
+
+            const auto* type = data_type.get();
+            if (type->is_nullable()) {
+                type = assert_cast<const DataTypeNullable*, TypeCheckOnRelease::DISABLE>(type)
+                               ->get_nested_type()
+                               .get();
+            }
+            int64_t value = 0;
+            WhichDataType which(type);
+            if (which.idx == TypeIndex::Int8) {
+                value = assert_cast<const ColumnInt8*, TypeCheckOnRelease::DISABLE>(column)
+                                ->get_element(0);
+            } else if (which.idx == TypeIndex::Int16) {
+                value = assert_cast<const ColumnInt16*, TypeCheckOnRelease::DISABLE>(column)
+                                ->get_element(0);
+            } else if (which.idx == TypeIndex::Int32) {
+                value = assert_cast<const ColumnInt32*, TypeCheckOnRelease::DISABLE>(column)
+                                ->get_element(0);
+            }
+            if (value <= 0) {
+                throw Exception(ErrorCode::INVALID_ARGUMENT,
+                                "The parameter cannot be less than or equal to 0.");
+            }
+            return value;
+        };
+
+        const auto& data_types = this->get_argument_types();
+        if (ArgsSize == 1) {
+            _threshold = std::min(get_param(_column_size, data_types, columns), (uint64_t)1000);
+        } else if (ArgsSize == 2) {
+            _threshold = std::min(get_param(_column_size, data_types, columns), (uint64_t)1000);
+            _reserved =
+                    std::min(std::max(get_param(_column_size + 1, data_types, columns), _threshold),
+                             (uint64_t)1000);
+        }
+
+        if (_threshold == 0 || _reserved == 0 || _threshold > 1000 || _reserved > 1000) {
+            throw Exception(ErrorCode::INTERNAL_ERROR,
+                            "approx_topn param error, _threshold: {}, _reserved: {}", _threshold,
+                            _reserved);
+        }
+
+        _init_flag = true;
+    }
+
+    mutable size_t _column_size = 0;
+    mutable bool _init_flag = false;
+    mutable uint64_t _threshold = 10;
+    mutable uint64_t _reserved = 300;
```

Comment:
seems those variables could put in AggregateFunctionTopKGenericData,
and implement some function more clearly.

Reply:
1. remove the use of optimization for is_aggregate_function_multi_top_with_null_variadic_inline.
2. These parameters won’t be used in AggregateFunctionTopKGenericData, so it seems better to place them outside.